ambry-commons/src/main/java/com/github/ambry/compression/LZ4Compression.java

```diff
@@ -55,6 +55,15 @@ public class LZ4Compression extends BaseCompressionWithLevel {
   @Override
   public String getAlgorithmName() { return ALGORITHM_NAME; }
 
+  /**
+   * Some compression algorithms like ZStd are picky on the source and destination buffer types.
+   * In LZ4, any combination of Heap and Direct memory are acceptable, so return false.
+   */
+  @Override
+  public boolean requireMatchingBufferType() {
```

Comment:
How about always matching buffer type to avoid any confusion ?
if source is direct, then dst is direct.
If source is heapmem, then dst is heapmem

Reply:
I see, when will there be a memcopy ? 
heapmem to heapmem OR directmem to directmem ?